utils/imds.go

```diff
@@ -0,0 +1,45 @@
+package utils
+
+import (
+	"fmt"
+	"time"
+)
+
+const (
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+	IMDSv1Server      = "http://169.254.169.254"
+	DataConfigMapName = "aws-data"
+	CIDRKey           = "vpc-cidr"
+)
+
+func IMDSFetchIPv4CIDR(imdsServerAddr string) (string, error) {
+	// This method needs an instance mac address to get the VPC ipv4 CIDR.
+	var mac string
+	err := Retry(3, 1*time.Second, func() error {
+		var err error
+		endpoint := fmt.Sprintf("%s/latest/meta-data/mac", imdsServerAddr)
+		if mac, err = HTTPGetAndParseBody(endpoint); err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to determine mac address of instance")
+	}
+
+	var cidr string
+	err = Retry(3, 1*time.Second, func() error {
```

Comment:
We decided to  handle retry logic with backoff from the outside. Why do we need to introduce additional (up to 2 seconds) of sleep here?